src/wallet/rpcwallet.cpp

```diff
@@ -2032,9 +2032,12 @@ UniValue getwalletinfo(const UniValue& params, bool fHelp)
     if (pwalletMain->IsCrypted())
         obj.pushKV("unlocked_until", nWalletUnlockTime);
     obj.pushKV("paytxfee",      ValueFromAmount(payTxFee.GetFeePerK()));
-    uint256 seedFp = pwalletMain->GetHDChain().seedFp;
-    if (!seedFp.IsNull())
-         obj.pushKV("seedfp", seedFp.GetHex());
+    auto mnemonicChain = pwalletMain->GetMnemonicHDChain();
+    if (mnemonicChain.has_value())
+         obj.pushKV("seedfp", mnemonicChain.value().GetSeedFingerprint().GetHex());
+    auto legacyChain = pwalletMain->GetLegacyHDChain();
+    if (legacyChain.has_value())
+         obj.pushKV("legacy_seedfp", legacyChain.value().GetSeedFingerprint().GetHex());
```

Comment:
I think I'd rather remove `seedfp` and add `legacy_seedfp` and `mnemonic_seedfp`. Also, I think it might be okay to remove the legacy seed fingerprint entirely, if the legacy seed is no longer being used as the source of randomness for any new addresses?

Reply:
This seems fine, but I'd rather keep `legacy_seedfp`, because its presence indicates that there have been legacy addresses generated with the identified seed.